fsw/cfe-core/unit-test/es_UT.c

```diff
@@ -4793,14 +4795,38 @@ void TestAPI(void)
     /* Test successfully deleting a child task */
     ES_ResetUnitTest();
     CFE_ES_Global.TaskTable[1].RecordUsed = true;
-    CFE_ES_Global.TaskTable[1].AppId = 0;
-    CFE_ES_Global.AppTable[0].AppState = CFE_ES_AppState_RUNNING;
-    CFE_ES_Global.AppTable[0].TaskInfo.MainTaskId = 15;
-    OS_TaskCreate(&CFE_ES_Global.TaskTable[1].TaskId, NULL, NULL, NULL,
+    CFE_ES_Global.TaskTable[1].AppId = 15;
+
+    CFE_ES_Global.TaskTable[16].RecordUsed = true;
+    CFE_ES_Global.TaskTable[16].AppId = 15;
+    CFE_ES_Global.TaskTable[17].RecordUsed = true;
+    CFE_ES_Global.TaskTable[17].AppId = 15;
+    CFE_ES_Global.TaskTable[17].TaskId = 25;
+    CFE_ES_Global.TaskTable[18].RecordUsed = true;
+    CFE_ES_Global.TaskTable[18].AppId = 15;
+    CFE_ES_Global.AppTable[15].AppState = CFE_ES_AppState_RUNNING;
+    CFE_ES_Global.AppTable[15].TaskInfo.MainTaskId = 25;
+
+    OS_TaskCreate(&CFE_ES_Global.TaskTable[16].TaskId, NULL, NULL, NULL,
                   0, 0, 0);
+   
+    CFE_ES_GetAppInfo(&AppInfo,CFE_ES_Global.TaskTable[1].AppId);    
+    uint32 NumORegTasksTemp = CFE_ES_Global.RegisteredTasks;
+    uint32 NumOChildTastTemp1 = AppInfo.NumOfChildTasks;
```

Comment:
Wouldn't the "NumOfChildTasks" here be a predictable/consistent number based on the previous TaskTable config?  That is, I think we should know exactly how many child task entries are associated with the AppId because it was just set up earlier in the test sequence.  If we do have a specific number which is the correct answer based on an explicitly set config, then we should assert for that specific number.

Also as a general style thing I think we try to avoid instantiating local variables in the middle of the function.